aws/sdk/examples/lambda-invoke-function/src/main.rs

```diff
@@ -2,52 +2,139 @@
     println!("Region:      {:?}", &region);
+    println!("Lambda client version: {}", lambda::PKG_VERSION);
+    
 
     SubscriberBuilder::default()
         .with_env_filter("info")
         .with_span_events(FmtSpan::CLOSE)
         .init();
 
+    // The AWS SDK for Rust service clients can be instantiated in a few
+    // different ways. The way we're instantiating it here is to first build
+    // a Config struct, then pass the Config to the Client.
     let config = Config::builder().region(region).build();
-
     let client = Client::from_conf(config);
 
+    // we call the `invoke()` method on the client. The term 'invoke' is a bit
+    // overloaded in the industry, but here it refers to a method named 
+    // 'Invoke' as part of the AWS Lambda API.
+    // https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html
+    //
+    // The Invoke API accepts several arguments, but we're only going to
+    // include the required arguments in this example. The function name can
+    // be the human friendly name with or without a version alias. It can be a
+    // full ARN, or a partial ARN that includes the AWS account ID. following 
+    // are examples of each that will point to the same AWS resource.
+    //
+    // short name: 'example-function'
+    // short name with version alias: 'example-function:LATEST
+    // partial ARN: 072326518754:function:example-function
+    // full ARN: arn:aws:lambda:us-west-2:072326518754:function:example-function
+    //
+    // We are going to use the full ARN to prevent any ambiguity in which
+    // function will be invoked.
     match client
         .invoke()
-        .function_name("arn:aws:lambda:us-west-2:892717189312:function:my-rusty-func")
+        .function_name("arn:aws:lambda:us-west-2:072326518754:function:hello-python")
         .send()
         .await
     {
+        // If the API call returns without an error, the Lambda Invoke API
+        // returns a response object containing the following:
+        //
+        // StatusCode: The HTTP status code is in the 200 range for a
+        //   successful request
+        //
+        // ExecutedVersion: The version of the function that executed. When
+        //   you invoke a function with an alias, this indicates which version
+        //   the alias resolved to.
+        //
+        // FunctionError: If present, indicates that an error occurred during
+        //   function execution. Details about the error are included in the
+        //   response payload.
+        //
+        // LogResult: The last 4 KB of the execution log, which is base64
+        //    encoded.
+        //
+        // Payload: The response from the function, or an error object.
+        //
+        // For our example, we are just going to interact with the payload
+        // portion of the response
         Ok(resp) => {
-            println!("Response:");
-            println!("  {:?}", resp.payload);
+            match resp.payload {
```

Comment:
maybe an `if let` since I don't really care about the `_ => ()` case?

Reply:
Actually this is why your build is failing ;-) Clippy isn't allowing this

Reply:
`cargo clippy`
```
error: you seem to be trying to use `match` for destructuring a single pattern. Consider using `if let`
   --> examples/lambda-invoke-function/src/main.rs:95:13
    |
95  | /             match resp.payload {
96  | |                 // I'm sure there is some better idiomatic Rust way to do this
97  | |                 // but here we are. The payload response is a blob of binary
98  | |                 // data that we need to turn into something useful.
...   |
111 | |                 _ => ()
112 | |             }
    | |_____________^
    |
    = note: `-D clippy::single-match` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#single_match
help: try this
    |
95  |             if let Some(blob) = resp.payload {
96  |                 // I assign the contents of the blob to another variable
97  |                 // so that I can borrow it later without the compiler being
98  |                 // mad at me.
99  |                 let random_inner_crap = blob.into_inner();
100 |                 let s = str::from_utf8(&random_inner_crap).unwrap();
  ...

error: aborting due to previous error

error: could not compile `lambda-invoke-function`
```